### PR TITLE
extend snapshot capabilities

### PR DESCRIPTION
**Pull request by c0depwn**
- add snapshot status
- add GET snapshot endpoint

### CODE REVIEW

File: flow/snapshot.go [hunk 1]
```diff
@@ -2,25 +2,54 @@ package flow
...
+const (
+	SnapshotStatusKeyAvailable SnaphostStatusKey = "available"
+	SnapshotStatusKeyCreating  SnaphostStatusKey = "creating"
+	SnapshotStatusKeyError     SnaphostStatusKey = "error"
+)
+
+type Status struct {
+	Id   Id                `json:"id"`
+	Name string            `json:"name"`
+	Key  string `json:"key"`
```
Comment (thecodinglab) [inline flow/snapshot.go:28]: please reformat this so that all the tags have the same indentation

File: flow/snapshot.go [hunk 2]
```diff
@@ -2,25 +2,54 @@ package flow
...
+	switch snapshot.Status.Key {
+	case SnapshotStatusKeyAvailable:
+		return true, nil
+	case SnapshotStatusKeyCreating:
+		return false, nil
+	case SnapshotStatusKeyError:
+		return false, errors.New(fmt.Sprintf("Snapshot with id %d has errored state", snapshot.Id))
+	}
+	return false, errors.New(fmt.Sprintf("Snapshot with id %d has unknown status '%s'", snapshot.Id, snapshot.Status.Key))
+}
```
Comment (thecodinglab) [inline flow/snapshot.go:51]: why not simply use the following?

```suggestion
func (s *Snapshot) IsAvailable() bool {
    return s.Status == SnapshotStatusKeyAvailable
}
```

**Review by thecodinglab — CHANGES_REQUESTED** [review]

File: flow/snapshot.go (hunk 2 above)
Comment (c0depwn) [reply flow/snapshot.go:51]: Was thinking that we could extend this with other statuses which might evaluate to an available state thats why I added the switch case. For example if we would have a status for post-processing which won't prevent the snapshot from being used to create a new volume in the future but I guess we can add that when we need it.

Comment (thecodinglab) [reply flow/snapshot.go:51]: alright, yeah but then i'd suggest using a map like
```go
var availabilityMapping = map[SnaphostStatusKey]bool{
  SnapshotStatusKeyAvailable: true,
  SnapshotStatusKeyCreating: false,
}

func (s *Snapshot) IsAvailable() bool {
  return availabilityMapping[s.Status.Key]
}
```

**Comment by thecodinglab** [comment]
@c0depwn I just noticed this is the deprecated API client which should not be used anymore and is just kept for backwards compatibility. We should rather focus on extending the new API rather than the old one

**Pull request merged**